esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1451,6 +1457,25 @@ EspAuthState CEspHttpServer::handleUserNameOnlyMode(EspAuthRequest& authReq)
         }
         return authFailed;
     }
+    if (!isValidUserName(userNameIn))
+    {
+        //If the username comes from the GetUserName page, the page should has the urlCookie and the askUserLogin() should be called to get the username.
+        //Otherwise, the BasicAuthenticationChallenge should be sent.
```

Comment:
A small wording change:

//If the username comes from the GetUserName page, the page should have set the urlCookie and askUserLogin() should be called to get the username.
//Otherwise, the BasicAuthenticationChallenge should be sent.

Reply:
But isn't GetUserName where askUserLogin() redirects you to?  So sending them back to where they came from?

Reply:
The askUserLogin() redirects to the GetUserName which is where they came from.